File: VB/MvpvmNavigation/ViewModels/MainFormViewModel.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports DevExpress.Mvvm.DataAnnotations
Imports MvpvmNavigation.Model
Imports DevExpress.Mvvm.POCO

Namespace MvpvmNavigation.ViewModels
	Public Class MainFormViewModel
		Private privateSelectedModuleType As ModuleType
		Public Overridable Property SelectedModuleType() As ModuleType
			Get
				Return privateSelectedModuleType
			End Get
			Set(ByVal value As ModuleType)
				privateSelectedModuleType = value
			End Set
		End Property
		Protected Overridable Sub OnSelectedModuleTypeChanged()
            Me.RaiseCanExecuteChanged(Sub(x) x.SelectModule(ModuleType.None))
			RaiseSelectedModuleTypeChanged()
		End Sub

		Public Event SelectedModuleTypeChanged As EventHandler

		Private Sub RaiseSelectedModuleTypeChanged()
			Dim handler = SelectedModuleTypeChangedEvent
			If handler IsNot Nothing Then
				handler(Me, EventArgs.Empty)
			End If
		End Sub

		<Command(UseCommandManager := False)> _
		Public Sub SelectModule(ByVal type As ModuleType)
			SelectedModuleType = type
		End Sub
		Public Function CanSelectModule(ByVal type As ModuleType) As Boolean
			Return type <> SelectedModuleType
		End Function
	End Class
End Namespace

File: CS/MvpvmNavigation/ViewModels/MainFormViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using DevExpress.Mvvm.DataAnnotations;
using MvpvmNavigation.Model;
using DevExpress.Mvvm.POCO;

namespace MvpvmNavigation.ViewModels {
    public class MainFormViewModel {
        public virtual ModuleType SelectedModuleType {
            get;
            set;
        }
        protected virtual void OnSelectedModuleTypeChanged() {
            this.RaiseCanExecuteChanged(x => x.SelectModule(ModuleType.None));
            RaiseSelectedModuleTypeChanged();
        }

        public event EventHandler SelectedModuleTypeChanged;

        void RaiseSelectedModuleTypeChanged() {
            var handler = SelectedModuleTypeChanged;
            if (handler != null) handler(this, EventArgs.Empty);
        }

        [Command(UseCommandManager = false)]
        public void SelectModule(ModuleType type) {
            SelectedModuleType = type;
        }
        public bool CanSelectModule(ModuleType type) {
            return type != SelectedModuleType;
        }
    }
}
